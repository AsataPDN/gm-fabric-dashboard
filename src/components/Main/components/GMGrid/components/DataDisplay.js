import styled from "styled-components";
import { PropTypes } from "prop-types";

import {
  FONT_STACK_DATA,
  FONT_SIZE_LG,
  COLOR_CONTENT_BACKGROUND,
  BORDER_RADIUS_BASE
} from "style/styleVariables";
import { spacingScale, contrastColor } from "style/styleFunctions";

const DataDisplay = styled.div`
  height: 100%;
  font-family: ${FONT_STACK_DATA};
  font-size: ${FONT_SIZE_LG};
  background-color: ${contrastColor(COLOR_CONTENT_BACKGROUND, 0.04).string()};
  border-radius: ${BORDER_RADIUS_BASE};
  padding: ${spacingScale(1)};

  padding-top: ${spacingScale(6)};
  ${props => (props.table ? "text-align: right" : "")};
`;

DataDisplay.propTypes = {
  table: PropTypes.bool
};

export default DataDisplay;
