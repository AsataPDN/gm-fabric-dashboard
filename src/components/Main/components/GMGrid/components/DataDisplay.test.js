import React from "react";
import { shallow } from "enzyme";
import DataDisplay from "./DataDisplay";

describe("DataDisplay", () => {
  it("matches snapshot when table is false", () => {
    const aDataDisplay = shallow(<DataDisplay table={false} />);
    expect(aDataDisplay).toMatchSnapshot();
  });
  it("matches snapshot when table is true", () => {
    const aDataDisplay = shallow(<DataDisplay table={true} />);
    expect(aDataDisplay).toMatchSnapshot();
  });
});
