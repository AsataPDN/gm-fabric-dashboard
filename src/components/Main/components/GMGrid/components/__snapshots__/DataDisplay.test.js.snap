// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataDisplay matches snapshot when table is false 1`] = `
.c0 {
  background-color: rgb(245,245,245);
  border-radius: 3px;
  font-family: Rubik,-apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen","Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue",sans-serif;
  font-size: 18px;
  height: 100%;
  padding: 8px;
  padding-top: 48px;
}

<div
  className="c0"
/>
`;

exports[`DataDisplay matches snapshot when table is true 1`] = `
.c0 {
  background-color: rgb(245,245,245);
  border-radius: 3px;
  font-family: Rubik,-apple-system,BlinkMacSystemFont,"Segoe UI","Roboto","Oxygen","Ubuntu","Cantarell","Fira Sans","Droid Sans","Helvetica Neue",sans-serif;
  font-size: 18px;
  height: 100%;
  padding: 8px;
  padding-top: 48px;
  text-align: right;
}

<div
  className="c0"
/>
`;
